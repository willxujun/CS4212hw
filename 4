// Source: Zhaoying e5.pl

class Main {
Void main(Int i, Int a, Int b,Int d){
Int t1;
Int t2;

Compute help;
while(true){
   t1 = 1*2;
}
}
}
class Dummy {
Compute c;
Int i;
Dummy j;

Int dummy() {
Bool i;
Bool j;
if (i || j) {
println("Square of d larger than sum of squares");
return 1;
}
else {
while(i) {
i = !j;
}
c = this.getCompute();
}
return this.getCompute().square(-3);
return i ;
}
Compute getCompute() {
// c = new Compute();
return c;
}
}
