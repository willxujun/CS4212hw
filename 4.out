Warning: main class should have exactly 1 method
Error: Return statement does not have appropriate type
In method main
In class None

Program
  Class None
    VariableList
    MethodList
      Void main
        FormalList
        VariableList
        StatementList
          Return
            (Int 0)
      Void other
        FormalList
        VariableList
        StatementList
          Return
            (Int 0)
  Class VarOnly
    VariableList
      Int a
    MethodList
  Class MethodOnly
    VariableList
    MethodList
      String toString
        FormalList
        VariableList
        StatementList
          Dispatch
            Access toString
              (Id a)
            ExpressionList

Semantics error detected. Refer to msgs and parse tree.
