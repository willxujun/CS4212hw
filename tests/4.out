===== CData3 ===== 
class MainC
class Functional
    Int a

===== CMtd3 ===== 
Void %MainC_0 [MainC this]
[Functional fo, Int i, Int j, Bool _t0, Functional _t1, Int _t2]
    readln i;
    _t0 = i > 0;
    if _t0 goto 0;
    goto 1;
    L0:
    _t1 = new Functional();
    fo = _t1;
    _t2 = %Functional_2 [fo, i];
    j = _t2;
    println j;
    L1:
    println "Error";
    return;

Int %Functional_2 [Functional this, Int b]
[Int _t3]
    _t3 = 3;
    return _t3;

Bool %Functional_1 [Functional this, Int input]
[Int a, Int b, Int _t4, Int _t5, Bool _t6]
    _t4 = input * 5;
    a = _t4;
    _t5 = input + 7;
    b = _t5;
    _t6 = a >= b;
    return _t6;

Bool %Functional_0 [Functional this]
[Bool _t7]
    _t7 = %Functional_1 [this, 7];
    return _t7;


